lib/OpenQA/Schema/ResultSet/Assets.pm

```diff
@@ -73,44 +73,32 @@ sub scan_for_untracked_assets {
     my ($self) = @_;
 
     # search for new assets and register them
-    for my $type (qw(iso repo hdd)) {
-        my $dh;
-        next unless opendir($dh, $OpenQA::Utils::assetdir . "/$type");
-        my %assets;
+    for my $type (TYPES) {
         my @paths;
-        for my $file (readdir($dh)) {
-            unless ($file eq 'fixed' or $file eq '.' or $file eq '..') {
-                push(@paths, "$OpenQA::Utils::assetdir/$type/$file");
-            }
-        }
-        closedir($dh);
-        if (opendir($dh, $OpenQA::Utils::assetdir . "/$type" . "/fixed")) {
+
+        for my $subtype (qw(/ /fixed)) {
+            my $path = $OpenQA::Utils::assetdir . "/$type$subtype";
+            my $dh;
+            next unless opendir($dh, $path);
             for my $file (readdir($dh)) {
                 unless ($file eq 'fixed' or $file eq '.' or $file eq '..') {
-                    push(@paths, "$OpenQA::Utils::assetdir/$type/fixed/$file");
+                    push(@paths, "$path/$file");
                 }
             }
             closedir($dh);
         }
+
         my %paths;
         for my $path (@paths) {
 
             my $basepath = basename($path);
-            # very specific to our external syncing
-            next if $basepath =~ m/CURRENT/;
```

Comment:
then provide an alternative. Removing this special case is not an option - see https://progress.opensuse.org/issues/59103

Reply:
All this code did was ensure nothing would be deleted. I shouldn't have added the ticket as `Fixed` here since it makes more sense to do in a separate branch. See #2465 for my proposal on how to sort that out.